src/transforms/stdnames.jl

```diff
@@ -0,0 +1,69 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    StdNames(:spec)
+
+The transform that standardizes the column names. There are three different methods of standardization: uppercase, camelcase, and snakecase.
+They can be specified by the `spec`` argument (`:upper`, `:camel`, and `:snake`, respectively). Transform defaults to uppercase. 
+
+# Examples
+
+```julia
+StdNames()
+StdNames(:camel)
+```
+"""
+struct StdNames <: Stateless
+  spec::Symbol
+end
+
+StdNames() = StdNames(:upper)
+
+isrevertible(::Type{StdNames}) = true
+
+function apply(transform::StdNames, table)  
+  cols = Tables.columns(table)
+  oldnames = string.(Tables.columnnames(cols))
+  
+  spec = transform.spec
+
+  spec == :camel && (names = _camel.(oldnames))
+  spec == :snake && (names = _snake.(oldnames))
+  spec == :upper && (names = _upper.(oldnames))
+
+  newnames = _unique(names)
+  oldnew = Dict(zip(oldnames, newnames))
+
+  rtrans = Rename(oldnew)
+  newtable, rcache = apply(rtrans, table)
+
+  newtable, (rtrans, rcache)
+end
+
+function revert(::StdNames, newtable, cache)
+  rtrans, rcache = cache
+  revert(rtrans, newtable, rcache)
+end
+
+const delim = ['_', ' ']
+
+function _unique(names)
+  newnames = String[]
+  for name in names
+    n = name
+    while n ∈ newnames
+      n = string(n, "_")
+    end
+    push!(newnames, n)
+  end
+
+  newnames
+end
+
+_camel(name) = join(uppercasefirst.(split(strip(name, delim), delim)))
```

Comment:
Do we really the strip here?

Reply:
We need it for snake case, otherwise, for example, "_A" gets split into "" and "A" and when we combine them, we get "_A" again.
  
Not sure why added this for camel case, TBH.